src/login/logind-session.c

```diff
@@ -529,7 +522,13 @@ int session_load(Session *s) {
                 if (r < 0)
                         log_debug_errno(r, "Failed to parse leader PID of session: %s", leader);
                 else {
-                        r = session_set_leader(s, pid);
+                        _cleanup_(pidref_done) PidRef p = PIDREF_NULL;
+
+                        r = pidref_set_pid(&p, pid);
```

Comment:
instead of parse_pid() + pidref_set_pid() you can also use pidref_set_pidstr() which does this in one call